src/main.cpp

```diff
@@ -403,39 +403,287 @@ bool FullSyncWithDPORNodes()
+                
+                dProbAccumulator = 1 - ((1 - dProbAccumulator) * (1 - (dUnitStakeProbability * nCoinWeight)));
+                if (fDebug10) LogPrintf("ETTS debug: dProbAccumulator = %e", dProbAccumulator);
+            }
+
+        }
+        nDeltaTime = nTime - nTimePrev;
+        nThrows = nDeltaTime / (STAKE_TIMESTAMP_MASK + 1);
+        if (fDebug10) LogPrintf("ETTS debug: nThrows = %i", nThrows);
+        dCumulativeProbability = 1 - ((1 - dCumulativeProbability) * pow((1 - dProbAccumulator), nThrows));
+        if (fDebug10) LogPrintf("ETTS debug: dCumulativeProbability = %e", dCumulativeProbability);
+        
+        if(dCumulativeProbability >= dConfidence)
+            break;
+        
+        nTimePrev = nTime;
+    }
+
+    // If (dConfidence - dCumulativeProbability) > 0, it means we exited the negative Gantt chart area and the desired confidence level
+    // has not been reached. All of the eligible UTXO's are contributing probability, and this is the final dProbAccumulator value.
+    // If the loop above is degenerate (i.e. only the current time pass through), then dCumulativeProbability will be zero.
+    // If it was not degenerate and the positive reqions in the Gantt chart area contributed some probability, then dCumulativeProbability will
+    // be greater than zero. We must compute the amount of time beyond nTime that is required to bridge the gap between
+    // dCumulativeProbability and dConfidence. If (dConfidence - dCumulativeProbability) <= 0 then we overshot during the Gantt chart area,
+    // and we will back off by nThrows amount, which will now be negative.
+    if (fDebug10) LogPrintf("ETTS debug: dProbAccumulator = %e", dProbAccumulator);
+    
+    // Shouldn't happen because if we are down here, we are staking, and there have to be eligible UTXO's, but just in case...
+    if (dProbAccumulator == 0.0)
+    {
+        if (fDebug10) LogPrintf("ETTS debug: ERROR in dProbAccumulator calculations");
+        return result;
+    }
+    
+    if (fDebug10) LogPrintf("ETTS debug: dConfidence = %f", dConfidence);
+    // If nThrows is negative, this just means we overshot in the Gantt chart loop and have to backtrack by nThrows.
+    nThrows = (int64_t)((log(1 - dConfidence) - log(1 - dCumulativeProbability)) / log(1 - dProbAccumulator));
+    if (fDebug10) LogPrintf("ETTS debug: nThrows = %i", nThrows);
+    
+    nDeltaTime = nThrows * (STAKE_TIMESTAMP_MASK + 1);
+    if (fDebug10) LogPrintf("ETTS debug: nDeltaTime = %i", nDeltaTime);
+    
+    // Because we are looking at the delta time required past nTime, which is where we exited the Gantt chart loop.
+    result = nDeltaTime + nTime - nCurrentTime;
+    if (fDebug10) LogPrintf("ETTS debug: ETTS at %d confidence = %i", dConfidence, result);
+
+    // The old calculation for comparative purposes...
+    double oldETTS = 0;
+    
+    oldETTS = GetTargetSpacing(nBestHeight) * GetEstimatedNetworkWeight(nPoSInterval) / MinerStatus.WeightSum;
```

Comment:
Is this used except for debug? If move it inside the if.

Reply:
Nope. I should move it.

Reply:
Ok. Done!